frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -230,9 +262,23 @@ function AddNewRecord(props: AddNewRecordProps): ReactElement {
     context: {
       accountId: account?.id,
       isUserJavascriptCodeValid: validateUserJsCodeAsync,
+      data: data,
     },
   });
 
+  const collectionValue = form.watch('collection');
+  const keyValue = form.watch('key');
+
+  useEffect(() => {
+    const validateKeyField = async () => {
+      form.clearErrors('key'); // clear the errors on the key field - this is to make sure that the errors are cleared if the user switches collection values
+      await form.trigger('key'); // trigger validation
+    };
+    if (collectionValue !== '' && keyValue !== '') {
+      validateKeyField();
+    }
+  }, [collectionValue, keyValue]);
```

Comment:
Instead of doing this, you should override the onChange handler on the collection input to simply reset the errors on the `key` whenever the collection changes.

That is more reacting and does not require this useEffect.

However, you'll probably want to take advantage of the `isDuplicate` function I've mentioned a few times throughout this PR to ensure that the collection that is being switched to does not have a duplicate `key` that might be in the form.